lib/scholar/naive_bayes/complement.ex

```diff
@@ -0,0 +1,440 @@
+    * `:classes` - class labels known to the classifier.
+
+    * `:feature_count` - Number of samples encountered for each (class, feature)
+        during fitting. This value is weighted by the sample weight when
+        provided.
+
+    * `:feature_all` - Number of samples encountered for each feature during fitting.
+        This value is weighted by the `sample_weights` when provided.
+
+  ## Examples
+
+      iex> x = Nx.iota({4, 3})
+      iex> y = Nx.tensor([1, 2, 0, 2])
+      iex> Scholar.NaiveBayes.Complement.fit(x, y, num_classes: 3)
+      %Scholar.NaiveBayes.Complement{
+        feature_log_probability: Nx.tensor(
+          [
+            [1.3062516450881958, 1.0986123085021973, 0.9267619848251343],
+            [1.2452157735824585, 1.0986123085021973, 0.9707789421081543],
+            [1.3499267101287842, 1.0986123085021973, 0.8979415893554688]
+          ]
+        ),
+        class_count: Nx.tensor([1.0, 1.0, 2.0]),
+        class_log_priors: Nx.tensor([-1.3862943649291992, -1.3862943649291992, -0.6931471824645996]),
+        classes: Nx.tensor([0, 1, 2]),
+        feature_count: Nx.tensor(
+          [
+            [6.0, 7.0, 8.0],
+            [0.0, 1.0, 2.0],
+            [12.0, 14.0, 16.0]
+          ]
+        ),
+        feature_all: Nx.tensor([18.0, 22.0, 26.0])
+      }
+      iex> x = Nx.iota({4, 3})
+      iex> y = Nx.tensor([1, 2, 0, 2])
+      iex> Scholar.NaiveBayes.Complement.fit(x, y, num_classes: 3, sample_weights: [1, 6, 2, 3])
+      %Scholar.NaiveBayes.Complement{
+        feature_log_probability: Nx.tensor(
+          [
+            [1.2953225374221802, 1.0986123085021973, 0.9343092441558838],
+            [1.2722758054733276, 1.0986123085021973, 0.9506921768188477],
+            [1.3062516450881958, 1.0986123085021973, 0.9267619848251343]
+          ]
+        ),
+        class_count: Nx.tensor([2.0, 1.0, 9.0]),
+        class_log_priors: Nx.tensor([-1.7917594909667969, -2.4849066734313965, -0.28768205642700195]),
+        classes: Nx.tensor([0, 1, 2]),
+        feature_count: Nx.tensor(
+          [
+            [12.0, 14.0, 16.0],
+            [0.0, 1.0, 2.0],
+            [45.0, 54.0, 63.0]
+          ]
+        ),
+        feature_all: Nx.tensor([57.0, 69.0, 81.0])
+      }
+  """
+  deftransform fit(x, y, opts \\ []) do
+    opts = NimbleOptions.validate!(opts, @opts_schema)
+
+    input_rank = Nx.rank(x)
+    targets_rank = Nx.rank(y)
+
+    if input_rank != 2 do
+      raise ArgumentError,
+            "wrong input rank. Expected x to be rank 2 got: #{input_rank}"
+    end
+
+    if targets_rank != 1 do
+      raise ArgumentError,
+            "wrong target rank. Expected target to be rank 1 got: #{targets_rank}"
+    end
+
+    {num_samples, _} = Nx.shape(x)
+    {num_targets} = Nx.shape(y)
+
+    if num_samples != num_targets do
+      raise ArgumentError,
+            "wrong input shape. Expected x to have the same first dimension as y, got: #{num_samples} for x and #{num_targets} for y"
+    end
```

Comment:
You can move this to `fit_n`. Please see the changes I published to the other NaiveBayes ones. :)